lib/chef/knife/ec_base.rb

```diff
@@ -121,6 +127,33 @@ def rest
         @rest ||= Chef::ServerAPI.new(server.root_url, {:api_version => "0"})
       end
 
+      def remote_users
+        @remote_users ||= rest.get('/users')
+      end
+
+      def remote_user_list
+        @remote_user_list ||= remote_users.keys.map(&:to_sym)
+      end
+
+      def local_user_list
+        @local_user_list ||= Dir.glob("#{@dest_dir}/users/*\.json").map { |u| File.basename(u, '.json').to_sym }
+      end
+
+      def for_each_user_purge
+        purge_list = if opt_parser.default_argv[1] == 'backup'
```

Comment:
This could be problematic if this code was ever called from a different context.  I'm wondering if it might be cleaner to have a `users_for_purge` function that could be implemented in the Backup and Restore classes (with an implementation here that raises an Unimplemented error.  I don't usually like that kind of OOP indirection but it feels a bit safe than looking at argv